gradle/native-targets.gradle

```diff
@@ -2,24 +2,78 @@
  * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 
+/**
+ * Specifies what subset of Native targets to build
+ *
+ * ALL — all possible for compilation
+ * HOST — host-specific ones, without cross compilation (e.g. do not build linuxX64 on macOS host)
+ * SINGLE — only for current OS (to import into IDEA / quickly run tests)
+ * DISABLED — disable all Native targets (useful with Kotlin compiler built from sources)
+ *
+ * For HOST mode, all targets are still listed in .module file, so HOST mode is useful for release process.
+ */
+enum NativeState { ALL, HOST, SINGLE, DISABLED }
+
+def getNativeState(String description) {
+    if (description == null) return NativeState.SINGLE
+    switch(description.toLowerCase()) {
+        case 'all':
+        case 'true':
+            return NativeState.ALL
+        case 'host':
+            return NativeState.HOST
+        case 'disabled':
+            return NativeState.DISABLED
+        // 'single', 'false', etc
+        default:
+            return NativeState.SINGLE
+    }
+}
+
 project.ext.ideaActive = System.getProperty('idea.active') == 'true'
-project.ext.deployMode = property('native.deploy') == 'true'
-project.ext.singleTargetMode = project.ext.ideaActive || !project.ext.deployMode
+project.ext.nativeState = getNativeState(property('native.deploy'))
+project.ext.singleTargetMode = project.ext.ideaActive || (project.ext.nativeState == NativeState.SINGLE)
 
 project.ext.nativeMainSets = []
 project.ext.nativeTestSets = []
 
+/**
+ * Disables compilation but leaves the target in .module file
+ */
+def disableCompilation(targets) {
+    configure(targets) {
+        compilations.all {
+            cinterops.all { project.tasks[interopProcessingTaskName].enabled = false }
```

Comment:
We don't have cinterop, do we?

Reply:
We don't. Although I've decided to left this snippet 'as-is' so it would not differ much with ktor's one